src/shims/io.rs

```diff
@@ -0,0 +1,160 @@
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+
+use crate::stacked_borrows::Tag;
+use crate::*;
+
+#[derive(Default)]
+pub struct FileHandler {
+    files: HashMap<i32, File>,
+    flags: HashMap<i32, i32>,
+    low: i32,
```

Comment:
i think 0, 1, and 2 need to be kept reserved, we already support writing to stdout and stderr.

Reply:
Oh shoo, that's true